### PR TITLE
factor out window raising method

### PR DESCRIPTION
**Pull request by sebasje**
The interesting change is moving a static function that only raises the dialog into its own class. This improves separation between syncing client and window system. The reason why I'm separating this one out is because it's going to be needed from my clientlib shared bits, such as the credentialstore, but would drag in the whole gui application, which would be defeating the purpose entirely.

This pull request also initializes an otherwise empty pointer, in a separate commit.

### CODE REVIEW

**Comment by sebasje** [comment]
The bot is wrong, I've signed and sent in the CLA.

**Comment by dragotin** [comment]
like it :+1:

File: src/gui/windowraiser.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class QWidget;
+
+namespace OCC {
+
+
+/**
+ * @brief The WindowRaiser class
+ * @ingroup gui
+ */
+class WindowRaiser
```
Comment (ogoffart) [inline src/gui/windowraiser.h]: This does not need to be a class.

We have lots of plain functions in the OCC namespace (for example: utility.h, filesystem.h)  so i think it's fine to have the raise dialog in a funciton without a class.

**Comment by ogoffart** [comment]
:+1:    only that it dooes not really need to be in a class but could be its own function

**Comment by guruz** [comment]
@sebasje Is this still something you'd want to do?

I also think it could just be a function with OCC:: namespace.

**Pull request closed without merging**